### PR TITLE
feat: adds base embeddable widget

### PR DESCRIPTION
**Pull request by mikaelbr**
Adds initial scaffold of widget, with some base features.

Can see widget documentation on http://localhost:3000/widget

Fixes part of https://github.com/AtB-AS/kundevendt/issues/14939

Includes:

- Autocomplete search
- Initial navigation to assistant / departures.
- Base form
- Install instructions
- Separate minimal build

Screenshot 2023-11-20 at 20 14 08

### CODE REVIEW

File: src/pages/widget/index.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+const initializeCode = html`<script>
+  window.PlannerWeb.init();
+</script>`;
+const outputCodeExample = html`<div id="planner-web"></div>
+  <script>
+    // Basic example of using dynamic output
+    document.querySelector('#planner-web').innerHTML = window.PlannerWeb.output;
+  </script> `;
+
+const AssistantPage: NextPage<WidgetPageProps> = (props) => {
```
Comment (adriansberg) [inline src/pages/widget/index.tsx:35]: Should rename this component.

File: src/pages/widget/index.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+            <h2>Installation</h2>
+
+            <p>
+              Install by copying HTML provided below. After loading JS and CSS
+              file it can be initialized using the following code:
+            </p>
+            <CopyMarkup content={initializeCode} />
+
+            <h3>Scripts (UMD / ESM)</h3>
```
Comment (adriansberg) [inline src/pages/widget/index.tsx:78]: Should this header be different?

**Comment by adriansberg** [comment]
There's some flashing of content it looks like when getting new results. Is there something that can be done with this?

2023-11-21 at 09 15 23

**Comment by adriansberg** [comment]
A bit strange behaviour when navigating with keyboard, the page scrolls to the top of the selected element.

2023-11-21 at 09 21 15

**Comment by mikaelbr** [comment]
Oh yes. Been trying to avoid, but haven't found a solution yet. 😬

**Review by adriansberg — APPROVED** [review]
All in all this looks great. A great start, and we can build upon this foundation. 🚀

**Pull request merged**